src/writer.jl

```diff
@@ -0,0 +1,309 @@
+        rec_start = bytes
+        calc_rec_bytes = sizeof(Int32) # shape code
+
+        # Write record number
+        bytes += Base.write(io, bswap(Int32(num)))
+
+        if ismissing(geom)
+            bytes += Base.write(io, bswap(Int32(calc_rec_bytes ÷ 2)))
+            rec_bytes += Base.write(io, SHAPECODE[Missing])
+
+            # Indices for .shx file
+            offset = rec_start ÷ 2
+            rec_len = rec_bytes ÷ 2 # 16 bit words
+            push!(shx_indices, IndexRecord(offset, rec_len))
+
+            # Update bytes
+            bytes += rec_bytes
+            continue
+        end
+
+        # Calculate record size
+        if trait isa GI.PointTrait 
+            calc_rec_bytes += sizeof(Float64) * 2 # point values
+            if hasz
+                calc_rec_bytes += sizeof(Float64) # z values
+            end
+            if hasm
+                calc_rec_bytes += sizeof(Float64) # measures
+            end
+        else
+            n = GI.npoint(geom)
+            calc_rec_bytes += sizeof(Rect) # Rect
+            calc_rec_bytes += sizeof(Int32) # num points
+            calc_rec_bytes += sizeof(Float64) * n * 2 # point values
+            if hasz
+                calc_rec_bytes += sizeof(Interval) # z interval
+                calc_rec_bytes += sizeof(Float64) * n # z values
+            end
+            if hasm
+                calc_rec_bytes += sizeof(Interval) # m interval
+                calc_rec_bytes += sizeof(Float64) * n # measures
+            end
+            numparts = _nparts(trait, geom) 
+            if !isnothing(numparts)
+                calc_rec_bytes += sizeof(Int32) # num parts
+                calc_rec_bytes += sizeof(Int32) * numparts # parts offsets
+            end
+        end
+
+        # length: measured in 16 bit increments, so divid by 2
+        bytes += Base.write(io, bswap(Int32(calc_rec_bytes ÷ 2)))
+        # code
+        rec_bytes += Base.write(io, shapecode)
+        # geometry
+        geom_bytes, geom_mbr, geom_zrange, geom_mrange = _write(io, trait, geom)
+        rec_bytes += geom_bytes
+
+        if first_geom
+            first_geom = false
+            mbr = geom_mbr
+            zrange = geom_zrange
+            mrange = geom_mrange
+        else
+            mbr = union(mbr, geom_mbr)
+            zrange = union(zrange, geom_zrange)
+            mrange = union(mrange, geom_mrange)
+        end
+
+        @assert rec_bytes == calc_rec_bytes
+
+        # Indices for .shx file
+        offset = rec_start ÷ 2 # 16 bit words
+        rec_len = rec_bytes ÷ 2 # 16 bit words
+        push!(shx_indices, IndexRecord(offset, rec_len))
+
+        # Record Header
+        # 32-bits (4 bytes) Record Number + 32-bits (4 bytes) Content length = 8 bytes
+        bytes += rec_bytes
+    end
+
+    # @assert bytes == content_size + header_bytes "The hardcoded accumulation of bytes $(content_size + header_bytes) should match how much bytes it has written into the buffer $bytes"
```

Comment:
Good, why comment it out?

Reply:
Doesn't actually make sense in practice, will delete

Reply:
There is now a per shape size check instead, and the whole file size just uses the actual written byte length so there is nothing to check.